.github/workflows/build.yml

```diff
@@ -130,6 +130,18 @@ jobs:
               comment_id: +COMMENT_ID,
             } );
 
+      - name: Trigger TeamCity build
+        if: ${{ fromJSON(steps.changed.outputs.projects)['plugins/jetpack'] == true }}
+        env:
+          SECRET: ${{ secrets.JETPACK__GITHUB_TEAMCITY_TOKEN }}
+          PR: ${{ github.event.number }}
+        run: |
+          curl -v --fail -L \
+            --url "https://mc.a8c.com/github-post-push-callback.php?jetpack-tc-trigger" \
+            --form-string "pr-number=$PR" \
+            --form-string "sha=$GITHUB_SHA" \
```

Comment:
Hm, apparently, $GITHUB_SHA is not what we need. For the latest commit, it used `40dfde5e762fd683d619027a4d6c727284910b1e` which isn't a part of this PR:

https://github.com/Automattic/jetpack/commit/40dfde5e762fd683d619027a4d6c727284910b1e

Reply:
For a PR event, I think you'll need to use `github.event.pull_request.head.sha`, like https://github.com/Automattic/jetpack/blob/test-tc-integration/.github/workflows/linting.yml#L284 does.

GITHUB_SHA should work for a push event.

Reply:
Oh, right. I guess we are not interested in push events, so I'll use the suggested approach without determining which event it is.

Reply:
We are interested in push events. That's the type of event triggered when a PR is merged into trunk.